src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.Android.cs

```diff
@@ -272,7 +272,7 @@ private Java.Lang.ICharSequence GetTextFormatted()
 			{
 				return EmptyString;
 			}
-			else if (UseInlinesFastPath)
+			else if (UseInlinesFastPath && FeatureConfiguration.IsJavaStringCachedEnabled)
```

Comment:
This will likely make the performance drop significantly when disabled. Let's make the modification in `GetNativeString` that will return the string immediately if the cache disabled.